Monika-After-Story/MonikaModDev
# Bare shoulder exprop correction and allowing for multiple required props

# Key Changes
* `required-hair-prop` and `required-clothes-prop` exprops now can accept a list of strings - basically ACS and hair can now require multiple exprops
* `bare-right-shoulder` - this exprop now actually means bare shoulder and not just compatibility with the `downtiedstrand` hair
* `bare-left-shoulder` (BLS) and `bare-shoulders` (BS) exprops added, same purpose as `bare-right-shoulder`
* `compat-downtiedstrand` (C_DTS) - new exprop that marks that a clothing item has compatible with the downtiestrand hair style

### exprop defaulting
* `default_exprops` - new overridable function on sprite object classes. This should be used to set additional exprops based on the ones currently set. This is called in the sprite object base constructor.

Currently the following defaults will happen:
* if `bare-shoulders` is set, the left and right bare shoulder exprops will be added
* if both the left and right bare shoulder exprops are added, `bare-shoulders` will be added
* if `bare-right-shoulder` is set, `compat-downtiedstrand` will be added

### exprop formatting
* `format_exprops` - new overridable function on sprite object classes. this should be used to format exprop values so they are in the correct format. This is called in the sprite object base constructor.

Currently the following formats will happen:
* if `required-hair-prop` is a string, it will be wrapped with a list
* if `required-clothes-prop` is a string, it will be wrapped with a list

### sprite object changes
* `mas_hair_downtiedstrand` - now requires the compat dts prop instead of BRS
* `mas_clothes_blackdress` - replaced BRS with BS
* `mas_clothes_blazerless` - replaced BRS with C_DTS
* `mas_clothes_marisa` - replaced BRS with C_DTS
* `mas_clothes_santa_lingerie` - replaced BRS with BS
* `mas_clothes_dress_newyears` - replaced BRS with BS
* `mas_clothes_sundress_white` - added BLS, which means this has both BRS and BLS set. (**NOTE: this is for ease of testing**)
* `mas_clothes_vday_lingerie` - replaced BRS with BS

### spritepack changes
* Black and white striped pullover - replaced BRS with BS
* Miku outfit - replaced BRS with BS
* Shell bikini - replaced BRS with BS
* Shoulderless sweater - replaced BRS with BS
* Pink Shirt - replaced BRS with BS
* Happy Shirt (Put on a happy face shirt) - replaced BRS with BS
* Green dress - replaced BRS with BS
* Tank top - replaced BRS with BS

**NOTE:** other spritepacks don't *need* to change to BS or C_DTS. BRS will work fine for now, but if something is released that utilizes actual bare shoulders, then spritepacks should update their jsons to be more accurate. 

# Testing
* currently nothing has been written to test multiple required props, but testing if the current behavior is still correct should be fine for now (aka, verify that whatever currently uses RQCP and RQHP still work as is). We can test an actual instance of using multiple required props when we have that as a sprite object.
* verify the RQCP and RQHP exprops are properly formatted. this can be done by opening console and checking the `ex_props` property on a sprite object. any instance of `required-clothes-prop` and `required-hair-prop` should be a list instead of a single string
* verify exprops are defaulting correctly. This can be checked by opening console and  checking the `ex_props` property on a sprite object. See above for the rules for bare shoulders and C_DTS. The sundress is the only instance where BLS+BRS is used instead of BS.
* verify the changed sprite objects still work with the `downtiedstrand` hair
* verify the changed spritepacks still work with the `downstiedstrand` hair
